MappedBytes has capacity >2G leading to ArithmeticException in NativeBytes methods

see #426 

```
[main/init_core-event-loop] WARN software.chronicle.fix.staticcode.FixTcpHandler - 
net.openhft.chronicle.core.io.IORuntimeException: java.lang.ArithmeticException: Unsigned Int 31-bit 5368708935 out of range
        at net.openhft.chronicle.wire.Demarshallable.newInstance(Demarshallable.java:62)
        at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.demarshallable(BinaryWire.java:3643)
        at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.typedMarshallable0(BinaryWire.java:3401)
        at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.typedMarshallable(BinaryWire.java:3371)
        at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue$StoreSupplier.acquire(SingleChronicleQueue.java:1127)
        at net.openhft.chronicle.queue.impl.WireStorePool.acquire(WireStorePool.java:66)
        at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue.storeForCycle(SingleChronicleQueue.java:603)
        at net.openhft.chronicle.queue.impl.single.StoreTailer.cycle(StoreTailer.java:1118)
        at net.openhft.chronicle.queue.impl.single.StoreTailer.moveToIndexResult0(StoreTailer.java:661)
        at net.openhft.chronicle.queue.impl.single.StoreTailer.moveToIndexInternal(StoreTailer.java:737)
        at net.openhft.chronicle.queue.impl.single.StoreTailer.next0(StoreTailer.java:295)
        at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument0(StoreTailer.java:237)
        at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument(StoreTailer.java:212)
        at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument(StoreTailer.java:156)
        ...
Caused by: java.lang.ArithmeticException: Unsigned Int 31-bit 5368708935 out of range
        at net.openhft.chronicle.core.Maths.toUInt31(Maths.java:659)
        at net.openhft.chronicle.bytes.internal.NativeBytesStore.read(NativeBytesStore.java:842)
        at net.openhft.chronicle.bytes.AbstractBytes.read(AbstractBytes.java:1134)
        at net.openhft.chronicle.bytes.util.AbstractInterner.intern(AbstractInterner.java:95)
        at net.openhft.chronicle.bytes.util.AbstractInterner.intern(AbstractInterner.java:74)
        at net.openhft.chronicle.wire.BinaryWire.readSmallField(BinaryWire.java:629)
        at net.openhft.chronicle.wire.BinaryWire.readEvent(BinaryWire.java:618)
        at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueStore.<init>(SingleChronicleQueueStore.java:82)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at net.openhft.chronicle.wire.Demarshallable.newInstance(Demarshallable.java:57)
        ... 30 more
```

NativeBytesStore is really a MappedBytesStore

The mapped file is 5G sized so the limit of 5368708935 is probably correct. Looks like MappedBytes has to override read() or expose safeLimit to it.

--- a/src/main/java/net/openhft/chronicle/bytes/AbstractBytes.java
+++ b/src/main/java/net/openhft/chronicle/bytes/AbstractBytes.java
@@ -1130,7 +1130,7 @@ public abstract class AbstractBytes<U>
 
     @Override
     public long read(@NonNegative long offsetInRDI, byte[] bytes, @NonNegative int offset, @NonNegative int length) throws IllegalStateException {
-        int len = Math.min(length, Maths.toUInt31(readLimit() - offsetInRDI));
+        int len = Maths.toUInt31(Math.min(length, requireNonNegative(readLimit() - offsetInRDI)));
         return bytesStore.read(offsetInRDI, bytes, offset, len);
     }
 

--- a/src/main/java/net/openhft/chronicle/bytes/RandomDataInput.java
+++ b/src/main/java/net/openhft/chronicle/bytes/RandomDataInput.java
@@ -662,7 +662,7 @@ public interface RandomDataInput extends RandomCommon {
             throws IllegalStateException {
         requireNonNull(bytes);
         try {
-            int len = Math.min(length, Maths.toUInt31(requireNonNegative(readLimit() - offsetInRDI)));
+            int len = Maths.toUInt31(Math.min(length, requireNonNegative(readLimit() - offsetInRDI)));
             for (int i = 0; i < len; i++)
                 bytes[offset + i] = readByte(offsetInRDI + i);
             return len;

--- a/src/main/java/net/openhft/chronicle/bytes/internal/NativeBytesStore.java
+++ b/src/main/java/net/openhft/chronicle/bytes/internal/NativeBytesStore.java
@@ -839,7 +839,7 @@ public class NativeBytesStore<U>
     @Override
     public long read(@NonNegative long offsetInRDI, byte[] bytes, @NonNegative int offset, @NonNegative int length) {
         requireNonNull(bytes);
-        int len = Math.min(length, Maths.toUInt31(requireNonNegative(readLimit() - offsetInRDI)));
+        int len = Maths.toUInt31(Math.min(length, requireNonNegative(readLimit() - offsetInRDI)));
 
         memory.readBytes(this.address + translate(offsetInRDI), bytes, offset, len);
         return len;

